### ISSUE TITLE
Filter out internal APIs

### ISSUE THREAD

**Issue by taisbak** [MicrosoftDocs/feedback#1854]
In .NET API Browser public as well as internal APIs are all listed.

E.g., when [searching for "Include" in Entity Framework Core](https://docs.microsoft.com/en-us/dotnet/api/?view=efcore-2.1&term=Include), a lot of the entries are described as
"This API supports the Entity Framework Core infrastructure and is not intended to be used directly from your code. This API may change or be removed in future releases.",
which clutters the result unnecessarily.

I suggest a checkbox, like "include internal APIs" to check when I want internal APIs as well.
Default should be unchecked.

**Comment by mairaw** [comment]
We could also filter out from docs certain namespaces. @divega would that make sense for some of these EF Core namespaces like Microsoft.EntityFrameworkCore.Query.Internal?

**Comment by divega** [comment]
I agree that for the API browser as it is those "public-internal" APIs just add a lot of clutter in search results and makes the experience of browsing the API that customers are much more likely to want to use very clumsy. 

It is good that they are discoverable in Intellisense in the IDE and the boilerplate comment they contain is readily available there to explain the caveats. But finding them in the API browser pages doesn't in my opinion add much value.

Given that, I also would like the *.Internal namespaces not to be included in the API browser.

However, I remember from talking to @ajcvickers about this before, that he had a different perspective.

**Comment by ajcvickers** [comment]
@divega I didn't want to do anything that would make it easier for people to miss that these are internal APIs. But now we have the analyzer I think it's probably fine to remove them from the API browser.

**Comment by mairaw** [comment]
Do you need help setting that up @divega?

**Comment by divega** [comment]
Yes, absolutely. And we still need help getting things setup for newer versions of our packages to be published to API browsers. The latest published is EF Core 2.1.

**Comment by divega** [comment]
@smitpatel is going to look into publishing all API reference for EF Core, and can take over this.
